applications/hotornot/src/hon_rater.erl

```diff
@@ -95,6 +99,17 @@ get_rate_data(JObj, ToDID, FromDID, Rates) ->
             {'ok', rate_resp(Rate, JObj)}
     end.
 
+-spec maybe_add_resource_flag(kz_json:object(), ne_binary()) -> kz_proplist().
+maybe_add_resource_flag(JObj, AccountId) ->
+    case kapps_account_config:get_from_reseller(AccountId, ?APP_NAME, <<"filter_by_resource_id">>, 'false') of
+        'true' ->
+            case kz_json:get_value(<<"Resource-ID">>, JObj) of
+                'undefined' -> [];
+                ResourceId -> [ResourceId]
```

Comment:
@lazedo Sorry Luis, I'm afaraid I can't get you correctly. 
I'm just adding ResourceId as one more flag which is handled like all other flags at hon_util:matching_options/2
https://github.com/2600hz/kazoo/blob/master/applications/hotornot/src/hon_util.erl#L147

Looks like this:
RateOptions: [<<"721a97ab1628023536dccb2e1fc3aae7">>]
RouteOptions: [<<"fax">>,<<"iamoption">>,<<"721a97ab1628023536dccb2e1fc3aae7">>]

I believe using tuple like {filter_resource_id, ResourceId} will break the process.

I don't see any more ways to pass resource_id to hon_util:matching_rates/4 without arity raise.

Regards,

PS Looks like previous way was a bit more transparent